### PR TITLE
CB-6

### PR DESCRIPTION
**Pull request by cansibi**

### CODE REVIEW

**Comment by cansibi** [comment]
ready to review

**Comment by cansibi** [comment]
cb-56 included

**Comment by cansibi** [comment]
ready to review

File: src/pages/CreateAccount/CreateAccount.tsx [hunk 1]
```diff
@@ -1,13 +1,38 @@
-import React, { FC, useState, ChangeEvent, FormEvent } from 'react'
+import React, { ChangeEvent, FC, FormEvent, useState } from 'react'
```
Comment (LH-Song) [inline src/pages/CreateAccount/CreateAccount.tsx:1]:
```suggestion
import React, { FC, FormEvent, useState } from 'react'
```

File: src/pages/CreateAccount/CreateAccount.tsx [hunk 2]
```diff
@@ -1,13 +1,38 @@
...
 const CreateAccount: FC = () => {
-  const [name, setName] = useState('')
+  const [name, setname] = useState('')
   const [email, setEmail] = useState('')
   const [password, setPassword] = useState('')
+  const [passwordVisible, setPasswordVisible] = useState(false)
+  const [passwordError, setPasswordError] = useState('')
   const [message, setMessage] = useState<string>('')
 
+  const handleInputChange =
```
Comment (LH-Song) [inline src/pages/CreateAccount/CreateAccount.tsx:17]:
```suggestion
const inputClassName =
    'w-full border border-gray-300 rounded-md px-3 py-2 mt-1 focus:outline-none focus:ring-1 focus:ring-gray-400'
    
  const handleInputChange =
```

File: src/pages/CreateAccount/CreateAccount.tsx [hunk 3]
```diff
@@ -1,13 +1,38 @@
...
+    (setValue: React.Dispatch<React.SetStateAction<string>>, validatePassword = false) =>
```
Comment (LH-Song) [inline src/pages/CreateAccount/CreateAccount.tsx:18]:
```suggestion
    (setValue: React.Dispatch<React.SetStateAction<string>>,) =>
```

File: src/pages/CreateAccount/CreateAccount.tsx [hunk 4]
```diff
@@ -1,13 +1,38 @@
...
+
+      if (validatePassword) {
+        const { value } = e.target
+        if (value.length < 8 || !/\d/.test(value) || !/[a-zA-Z]/.test(value)) {
+          setPasswordError(' At least 8 characters with 1 letter')
+        } else {
+          setPasswordError('')
+        }
+      }
+    }
```
Comment (LH-Song) [inline src/pages/CreateAccount/CreateAccount.tsx:30]:
```suggestion
     let errorMessage = ''

      if (value.length < 6 || !/\d/.test(value) || !/[a-zA-Z]/.test(value)) {
        errorMessage = 'At least 6 characters with 1 letter'
      }

      setPasswordError(value !== '' ? errorMessage : '')
      
    }
```

File: src/pages/CreateAccount/CreateAccount.tsx [hunk 5]
```diff
@@ -28,24 +53,69 @@ const CreateAccount: FC = () => {
...
+            className="w-full border border-gray-300 rounded-md px-3 py-2 mt-1 focus:outline-none focus:ring-1 focus:ring-gray-400"
+          />
+          <div className="mt-4 text-xs font-bold">Password</div>
+          <Input
+            id="password"
+            className="w-full border border-gray-300 rounded-md px-3 py-2 mt-1 focus:outline-none focus:ring-1 focus:ring-gray-400 focus:ring-opacity-50"
+            type={passwordVisible ? 'text' : 'password'}
+            value={password}
+            placeholder="********"
+            onChange={handleInputChange(setPassword, true)}
```
Comment (LH-Song) [inline src/pages/CreateAccount/CreateAccount.tsx:88]:
```suggestion
              onChange={handleInputChange(setPassword)}
```

File: src/pages/CreateAccount/CreateAccount.tsx [hunk 6]
```diff
@@ -28,24 +53,69 @@ const CreateAccount: FC = () => {
...
+                  {passwordVisible ? <Visibility /> : <VisibilityOff />}
+                </IconButton>
+              </InputAdornment>
+            }
+          />
+          <div>
+            <Typography className="caption" color="error">
+              {passwordError}
+            </Typography>
+          </div>
```
Comment (LH-Song) [inline src/pages/CreateAccount/CreateAccount.tsx:101]:
```suggestion
     {passwordError && (
              <Typography className="caption" color="error">
                {passwordError}
              </Typography>
            )}
```

**Review by LH-Song — CHANGES_REQUESTED** [review]
👖

**Review by LH-Song — APPROVED** [review]
LGTM🚀

**Pull request merged**